flair/datasets/sequence_labeling.py

```diff
@@ -443,6 +434,167 @@ def __init__(
+        """Generates a label for each token in the sentence in BIO format. This function requires that the labels corresponding to the label_type are span-level tokens.
+
+        Args:
+            sentence: a flair sentence to generate labels for
+            label_type: a string representing the type of the labels, e.g., "ner"
+        """
+        list_of_labels = ["O" for _ in range(len(sentence.tokens))]
+        for label in sentence.get_labels(label_type):
+            tokens = label.data_point.tokens
+            start_token_index = tokens[0]._internal_index
+            list_of_labels[start_token_index - 1] = f"B-{label.value}"
+            for token in tokens[1:]:
+                token_index = token._internal_index
+                list_of_labels[token_index - 1] = f"I-{label.value}"
+        return list_of_labels
+
+    @staticmethod
+    def write_dataset_to_file(
+        dataset: Dataset, file_path: Path, label_type_tuples: List[tuple], column_delimiter: str = "\t"
+    ) -> None:
+        """Writes a dataset to a file.
+
+        Following these two rules:
+        (1) the text and the label(s) of every token is represented in one line separated by column_delimiter
+        (2) every sentence is separated from the previous one by an empty line
+        """
+        with open(file_path, mode="w") as output_file:
+            for sentence in _iter_dataset(dataset):
+                texts = [token.text for token in sentence.tokens]
+                texts_and_labels = [texts]
+                for label_type, level in label_type_tuples:
+                    if level is Token:
+                        texts_and_labels.append(ColumnCorpus.get_token_level_label_of_each_token(sentence, label_type))
+                    elif level is Span:
+                        texts_and_labels.append(ColumnCorpus.get_span_level_label_of_each_token(sentence, label_type))
+                    else:
+                        raise NotImplementedError(f"The level of {label_type} is neither token nor span.")
+
+                for text_and_labels_of_a_token in zip(*texts_and_labels):
+                    output_file.write(column_delimiter.join(text_and_labels_of_a_token) + "\n")
+                output_file.write("\n")
+
+    @classmethod
+    def load_corpus_with_meta_data(cls, directory: Path) -> "ColumnCorpus":
+        """Creates a ColumnCorpus instance from the directory generated by 'write_to_directory'."""
+        with open(directory / "meta_data.json") as file:
+            meta_data = json.load(file)
+
+        meta_data["column_format"] = {int(key): value for key, value in meta_data["column_format"].items()}
+
+        return cls(
+            data_folder=directory,
+            autofind_splits=True,
+            skip_first_line=False,
+            **meta_data,
+        )
+
+    def get_level_of_label(self, label_type: str) -> Union[Type[Token], Type[Span]]:
```

Comment:
This method has only one usage, namely in the `write_to_directory` method. Making it public to the `ColumnCorpus` might be confusing to users. You can declare it as internal by prefixing the method name with a "_".